sw/device/boot_rom/bootstrap.c

```diff
@@ -33,81 +39,97 @@ static int bootstrap_requested(void) {
   // Read pin
   uint32_t gpio_in;
   dif_gpio_all_read(&gpio, &gpio_in);
-  return !!(gpio_in & GPIO_BOOTSTRAP_BIT_MASK);
+  return gpio_in != 0 && GPIO_BOOTSTRAP_BIT_MASK != 0;
```

Comment:
`return gpio_in != 0 && GPIO_BOOTSTRAP_BIT_MASK != 0;` => `return (gpio_in & GPIO_BOOTSTRAP_BIT_MASK) != 0`.

Would be nice to do it in terms of `bitfield_get_bit32` (which doesn't exist yet).

Reply:
I really really hate how not-type-safe any of this mess is.